File: Week_09/14/longest_common_prefix.py
from typing import List


class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""

        ans = strs[0]
        idx = 0
        flag = True

        while flag and idx < len(ans):
            for s in strs[1:]:
                if idx > len(s) or s[idx] != ans[idx]:
                    flag = False
                    break
            else:
                idx += 1
        return ans[:idx]

File: Week_09/14/longest_common_prefix.go
package main

func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	ans := strs[0]
	idx := 0
out:
	for idx < len(ans) {
		for i := 1; i < len(strs); i++ {
			if idx >= len(strs[i]) || strs[i][idx] != ans[idx] {
				break out
			}
		}
		idx++
	}
	return ans[:idx]
}
